### PR TITLE
Implement Nested Medium Priority object control

### PR DESCRIPTION
**Pull request by Mango-3**
Medium Priority of the box object in Max is set to 2:
max

and the exported scene is correctly imported with the setting in appleseed studio:
studio

### CODE REVIEW

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp [hunk 1]
```diff
@@ -75,7 +75,8 @@ namespace
         ParamIdVisibilitySpecular       = 7,
         ParamIdVisibilitySSS            = 8,
         ParamIdSSSSet                   = 9,
-        ParamIdOptimizeForInstancing    = 10
+        ParamIdOptimizeForInstancing    = 10,
+        paramIdNestedmediumPriority     = 11
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: Capitalize *Param* and *Medium*: `ParamIdNestedMediumPriority`

Thinking of it, I think `ParamIdMediumPriority` (without *Nested*) would be better.

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp [hunk 2]
```diff
@@ -145,7 +146,10 @@ namespace
         ParamIdSSSSet, L"sss_set", TYPE_STRING, 0, IDS_SSS_SET,
             p_ui, ParamMapIdVisibility, TYPE_EDITBOX, IDC_SSS_SET,
         p_end,
-
+        paramIdNestedmediumPriority, L"nested_medium_priority", TYPE_INT, P_TRANSIENT, 0,
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: Similarly here, `medium_priority` ought to be enough.

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp [hunk 3]
```diff
@@ -352,6 +356,14 @@ std::string AppleseedObjPropsMod::get_sss_set(const TimeValue t) const
     return str_value != nullptr ? wide_to_utf8(str_value) : std::string();
 }
 
+int AppleseedObjPropsMod::get_medium_priority(const TimeValue t) const
+{
+   int int_value = 0;
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: You assign 0 then immediately overwrite the variable.

You can do it all in a single line:

    return m_pblock->GetInt(paramIdNestedmediumPriority, t, FOREVER);

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp [hunk 4]
```diff
@@ -352,6 +356,14 @@ std::string AppleseedObjPropsMod::get_sss_set(const TimeValue t) const
...
+   int_value = m_pblock->GetInt(paramIdNestedmediumPriority, t, FOREVER);
+   return int_value;
+}
+
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: Our rules for vertical spacing are:

- Always use a *single* blank line to separate things.
- *Except* before block comments (such as the one below) which should be preceded by *two* blank lines.

File: src/appleseed-max-impl/appleseedobjpropsmod/resource.h [hunk 5]
```diff
@@ -2,49 +2,43 @@
...
-
 #define IDC_BUTTON_VISIBILITY_SSS           7090
 #define IDS_VISIBILITY_SSS                  7091
 #define IDC_SSS_SET                         7092
 #define IDS_SSS_SET                         7093
 #define IDC_SSS_SET_LABEL                   7094
-
 #define IDC_BUTTON_OPTIMIZE_FOR_INSTANCING  7101
 #define IDS_OPTIMIZE_FOR_INSTANCING         7102
+#define IDS_STATIC_Nested_Medium_Priority   7200
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedobjpropsmod/resource.h]: Capitalize.

File: src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp [hunk 6]
```diff
@@ -470,6 +470,25 @@ namespace
         return std::string();
     }
 
+    int get_medium_priority(Object* object, const TimeValue time)
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp]: This looks good, however we now have four times the same piece of code that looks for a modifier of type `AppleseedObjPropsMod`. That can easily be refactored into a separate function. Want to give this a try? If so, you can do it in a later PR if you prefer.

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp (hunk 1 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: OK

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp (hunk 2 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: agree

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp (hunk 3 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: Changed it as you have shown.

File: src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp (hunk 4 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedobjpropsmod/appleseedobjpropsmod.cpp]: Ok

File: src/appleseed-max-impl/appleseedobjpropsmod/resource.h (hunk 5 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedobjpropsmod/resource.h]: Done

File: src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp (hunk 6 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp]: I will try it in another PR

**Pull request merged**